WordPress/src/test/java/org/wordpress/android/ui/activitylog/list/filter/ActivityLogTypeFilterViewModelTest.kt

```diff
@@ -1,51 +1,131 @@
 package org.wordpress.android.ui.activitylog.list.filter
 
+import com.nhaarman.mockitokotlin2.anyOrNull
+import com.nhaarman.mockitokotlin2.times
+import com.nhaarman.mockitokotlin2.verify
+import com.nhaarman.mockitokotlin2.verifyNoMoreInteractions
+import com.nhaarman.mockitokotlin2.whenever
 import kotlinx.coroutines.InternalCoroutinesApi
 import org.assertj.core.api.Assertions.assertThat
 import org.junit.Before
 import org.junit.Test
+import org.mockito.Mock
 import org.wordpress.android.BaseUnitTest
 import org.wordpress.android.TEST_DISPATCHER
 import org.wordpress.android.test
 import org.wordpress.android.ui.activitylog.list.filter.ActivityLogTypeFilterViewModel.ListItemUiState
 import org.wordpress.android.ui.activitylog.list.filter.ActivityLogTypeFilterViewModel.UiState
-import org.wordpress.android.ui.activitylog.list.filter.ActivityLogTypeFilterViewModel.UiState.Content
+import org.wordpress.android.ui.activitylog.list.filter.DummyActivityTypesProvider.DummyActivityType
+import org.wordpress.android.ui.activitylog.list.filter.DummyActivityTypesProvider.DummyAvailableActivityTypesResponse
 
 @InternalCoroutinesApi
 class ActivityLogTypeFilterViewModelTest : BaseUnitTest() {
     private lateinit var viewModel: ActivityLogTypeFilterViewModel
+    @Mock private lateinit var dummyActivityTypesProvider: DummyActivityTypesProvider
 
     @Before
     fun setUp() {
-        viewModel = ActivityLogTypeFilterViewModel(TEST_DISPATCHER, TEST_DISPATCHER)
+        viewModel = ActivityLogTypeFilterViewModel(dummyActivityTypesProvider, TEST_DISPATCHER, TEST_DISPATCHER)
     }
 
     @Test
     fun `fullscreen loading shown, when screen initialized`() = test {
-        val uiStates = initObservers().uiStates
+        val uiStates = init().uiStates
 
-        viewModel.start()
+        viewModel.start(0L)
 
         assertThat(uiStates[0]).isInstanceOf(UiState.FullscreenLoading::class.java)
     }
 
     @Test
-    fun `section header gets added as first item in the list`() {
-        val uiStates = initObservers().uiStates
+    fun `available activity types fetched, when screen initialized`() = test {
+        init()
 
-        viewModel.start()
+        viewModel.start(0L)
 
-        assertThat((viewModel.uiState.value as Content).items[0])
+        verify(dummyActivityTypesProvider).fetchAvailableActivityTypes(anyOrNull())
+        verifyNoMoreInteractions(dummyActivityTypesProvider)
+    }
+
+    @Test
+    fun `section header gets added as first item in the list`() = test {
+        init()
+
+        viewModel.start(0L)
+
+        assertThat((viewModel.uiState.value as UiState.Content).items[0])
                 .isInstanceOf(ListItemUiState.SectionHeader::class.java)
     }
 
-    private fun initObservers(): Observers {
+    @Test
+    fun `content shown, when fetch available activity types completes successfully`() = test {
+        val uiStates = init(successResponse = true).uiStates
```

Comment:
Minor (🔍 ): The `uiStates` is not used in this test, is it?

Reply:
Fixed in 4eda09587ac3a0703d0c0e09c58be08047141f23